packages/core/src/tracing/idleSpan.ts

```diff
@@ -150,19 +159,26 @@ export function startIdleSpan(startSpanOptions: StartSpanOptions, options: Parti
       // Ensure we end with the last span timestamp, if possible
       const spans = getSpanDescendants(span).filter(child => child !== span);
 
+      const spanJson = spanToJSON(span);
+
       // If we have no spans, we just end, nothing else to do here
-      if (!spans.length) {
+      // Likewise, if users explicitly ended the span, we simply end the span without timestamp adjustment
+      if (!spans.length || !trimIdleSpanEndTimestamp) {
         onIdleSpanEnded(spanEndTimestamp);
         return Reflect.apply(target, thisArg, [spanEndTimestamp, ...rest]);
       }
 
-      const childEndTimestamps = spans
-        .map(span => spanToJSON(span).timestamp)
-        .filter(timestamp => !!timestamp) as number[];
-      const latestSpanEndTimestamp = childEndTimestamps.length ? Math.max(...childEndTimestamps) : undefined;
+      // Otherwise, we adjust the end timestamp of the idle span to the latest span end timestamp of its children
+      const latestSpanEndTimestamp = spans?.reduce((acc: number | undefined, current) => {
+        const currentSpanJson = spanToJSON(current);
+        if (acc == null) {
+          return currentSpanJson.timestamp;
+        }
+        return Math.max(acc, currentSpanJson.timestamp || 0);
+      }, undefined);
```

Comment:
This was just a performance improvement since previously we iterated 3x over the `spans` and we only need to do this once. (Can also go for a good old for loop if anyone dislikes this reduction)